lib/rubocop/cop/money/missing_currency.rb

```diff
@@ -32,7 +32,8 @@ class MissingCurrency < Cop
         PATTERN
 
         def on_send(node)
-          money_new(node) do |_amount, currency_arg|
+          money_new(node) do |amount, currency_arg|
+            return if amount&.splat_type?
```

Comment:
`splat_type?` is this a method within rubocop AST?

Reply:
Yup! Every `Node` type responds to all the `X_type?` methods.

Using `ruby-parse -e "..code..."`, we can inspect the AST:

<table><tr><th>Code</th><th>AST</th><th><code>amount</code></th><th><code>currency_arg</code></th></tr><tr><td>

```ruby
Money.new
```

</td><td>

```ruby
(send
  (const nil :Money) :new)
```

</td><td>

```ruby
nil
```

</td><td>

```ruby
nil
```

</td></tr><tr><td>

```ruby
Money.new(1)
```

</td><td>

```ruby               
(send
  (const nil :Money) :new
  (int 1))
```

</td><td>

```ruby
(int 1)
```

</td><td>

```ruby
nil
```

</td></tr><tr><td>

```ruby
Money.new(1, "CAD")
```

</td><td>

```ruby
(send
  (const nil :Money) :new
  (int 1)
  (str "CAD"))
```

</td><td>

```ruby
(int 1)
```

</td><td>

```ruby
(str "CAD")
```

</td></tr><tr><td>

```ruby
amount_and_currency = [1, "CAD"]
Money.new(*amount_and_currency)
```

</td><td>

```ruby
(begin
  (lvasgn :amount_and_currency
    (array
      (int 1)
      (str "CAD")))
  (send
    (const nil :Money) :new
    (splat
      (lvar :amount_and_currency))))
```

</td><td>

```ruby
(splat
  (lvar :amount_and_currency))
```

</td><td>

```ruby
nil
```

</td></tr><tr><td>

</tr></table>

From there, we can see the only cases we care about flagging in this Cop are when
- `currency_arg` is `nil`, and
- `amount` is not a `splat` type node